### PR TITLE
CLOUDP-196825: Setup ux improvements

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

- Spinner can only go in the least line and not on the side
- Spinner is bold and cyan is the best colour I could find ([available colours](https://pkg.go.dev/github.com/briandowns/spinner#readme-set-or-change-the-color-of-the-spinner-default-color-is-white-the-spinner-will-need-to-be-restarted-to-pick-up-the-change))

_Jira ticket:_[CLOUDP-196825](https://jira.mongodb.org/browse/CLOUDP-196825)

### First time setup experience
https://github.com/mongodb/mongodb-atlas-cli/assets/40155621/5a8a8a44-e703-49bc-b001-2df2bf129279

### Other times experience (without images) / and then with images

https://github.com/mongodb/mongodb-atlas-cli/assets/40155621/92c341e1-7bc0-4b3f-815f-daa271a2254c

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -70,7 +70,7 @@ require (
 	github.com/AzureAD/microsoft-authentication-library-for-go v1.1.1 // indirect
 	github.com/BurntSushi/toml v1.3.2 // indirect
 	github.com/PaesslerAG/gval v1.0.0 // indirect
-	github.com/ProtonMail/go-crypto v0.0.0-20230217124315-7d5c6f04bbb8 // indirect
+	github.com/ProtonMail/go-crypto v0.0.0-20230528122434-6f98819771a1 // indirect
```
Comment (gssbzn) [inline go.mod:73]: why all these updates?

File: internal/cli/atlas/deployments/setup.go [hunk 2]
```diff
@@ -131,7 +169,74 @@ Creating your cluster %s [this might take several minutes]
...
+	}
+
+	steps, needPodmanSetup, needToPullImages := opts.planSteps(ctx)
+	currentStep := 1
+	longWaitWarning := ""
+	if steps > shortStepCount {
+		longWaitWarning = " [this might take several minutes]"
+	}
+
+	_, _ = log.Warningln(fmt.Sprintf("Creating your cluster %s%s", opts.DeploymentName, longWaitWarning))
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/setup.go:213]: why not
```suggestion
	_, _ = log.Warningf("Creating your cluster %s%s", opts.DeploymentName, longWaitWarning)
```
?

File: internal/cli/atlas/deployments/options/deployment_opts.go [hunk 3]
```diff
@@ -78,6 +78,14 @@ func (opts *DeploymentOpts) LocalMongoMetricsVolume() string {
 	return fmt.Sprintf("mongot-local-metrics-%s", opts.DeploymentName)
 }
 
+func (opts *DeploymentOpts) MongodDockerImageName() string {
+	return fmt.Sprintf("docker.io/mongodb/mongodb-enterprise-server:%s-ubi8", opts.MdbVersion)
+}
+
+func MongotDockerImageName() string {
+	return "docker.io/mongodb/apix_test:mongot-preview"
+}
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/options/deployment_opts.go:87]: seems like a conts more than a function

File: internal/cli/atlas/deployments/podman/client.go [hunk 4]
```diff
@@ -70,13 +98,45 @@ type Client interface {
...
+
+	err := Installed()
+	if err != nil {
+		d.Installed = false
+		return d
+	}
+
+	info, err := o.machineInspect(ctx)
+	if err != nil { // machine is already present
+		d.MachineFound = false
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/podman/client.go:123]: I find this a bit confusing, the comment says machine present but then machinefound is set to false?

File: internal/cli/atlas/deployments/podman/client.go [hunk 5]
```diff
@@ -114,11 +174,14 @@ func (o *client) machineStart(ctx context.Context) error {
...
 }
 
-func (o *client) Ready(ctx context.Context) error {
+func Installed() error {
 	if _, err := exec.LookPath("podman"); err != nil {
-		return ErrPodmanNotReady
+		return ErrPodmanNotFound
 	}
+	return nil
+}
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/podman/client.go:182]: this probably belong in the podman package

File: internal/cli/atlas/deployments/setup.go (hunk 2 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/setup.go:213]: also `\n` in the end

File: internal/cli/atlas/deployments/options/deployment_opts.go (hunk 3 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/options/deployment_opts.go:87]: agreed const

File: internal/cli/atlas/deployments/podman/client.go [hunk 6]
```diff
@@ -229,6 +297,29 @@ func (o *client) ListContainers(ctx context.Context, nameFilter string) ([]*Cont
 	return containers, err
 }
 
+func (o *client) ListImages(ctx context.Context, nameFilter string) ([]*Image, error) {
+	args := []string{"image", "list", "--format", "json"}
```
[the remark this answers is not in the document]
Comment (blva) [reply internal/cli/atlas/deployments/podman/client.go:301]: yes, we have a ticket to try to use it but only after we're demo ready  https://jira.mongodb.org/browse/CLOUDP-196392

File: go.mod (hunk 1 above)
Comment (blva) [reply go.mod:73]: inherited from another commit and had to run go mod tidy

File: internal/cli/atlas/deployments/podman/client.go (hunk 4 above)
Comment (blva) [reply internal/cli/atlas/deployments/podman/client.go:123]: good catch, the comment comes from copy paste, removing

File: internal/cli/atlas/deployments/podman/client.go (hunk 5 above)
Comment (blva) [reply internal/cli/atlas/deployments/podman/client.go:182]: I'm confused, it's under podman package now, no?

File: go.mod (hunk 1 above)
Comment (gssbzn) [reply go.mod:73]: can you roll these back? mod tidy doesn't produce anything for me

File: internal/cli/atlas/deployments/podman/client.go (hunk 5 above)
Comment (gssbzn) [reply internal/cli/atlas/deployments/podman/client.go:182]: oh sorry, I saw `internal/cli/atlas/deployments/` and that's cli logic, the popman package should be a top level package `internal/podman/client.go` it's not related to the commands

File: go.mod (hunk 1 above)
Comment (blva) [reply go.mod]: yep, done! thanks

File: internal/cli/atlas/deployments/podman/client.go (hunk 5 above)
Comment (blva) [reply internal/cli/atlas/deployments/podman/client.go:182]: ahh gotcha, let me try to refactor this quick and see if it's ok

File: internal/cli/atlas/deployments/setup.go [hunk 7]
```diff
@@ -139,7 +177,74 @@ Creating your cluster %s [this might take several minutes]
...
+}
+
+func (opts *SetupOpts) createLocalDeployment(ctx context.Context) error {
+	if err := podman.Installed(); err != nil {
+		return err
+	}
+
+	steps, needPodmanSetup, needToPullImages := opts.planSteps(ctx)
+	currentStep := 1
+	longWaitWarning := ""
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/setup.go:216]: I think for future work it would be good to investigate using a state machine lib

**Review by gssbzn — APPROVED** [review]
LGTM great work, comments are not blocking

File: internal/cli/atlas/deployments/setup.go [hunk 8]
```diff
@@ -639,6 +744,7 @@ func SetupBuilder() *cobra.Command {
 func (opts *SetupOpts) start() {
 	if opts.IsTerminal() {
 		opts.s = spinner.New(spinner.CharSets[9], spinnerSpeed)
+		_ = opts.s.Color("cyan", "bold")
```
[the remark this answers is not in the document]
Comment (blva) [reply internal/cli/atlas/deployments/setup.go:747]: i'd think worst case it just won't have any color. is there any terminal of that sort that i can test on mac?

[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/atlas/deployments/setup.go:747]: you should be able to test how this looks by down `export TERM=test-m` the important thing is adding `-m` at the end

[the remark this answers is not in the document]
Comment (blva) [reply internal/cli/atlas/deployments/setup.go:747]: cool, thanks, I just tested. Survey respects it, but this does not

**Pull request merged**